### ISSUE TITLE
Adaptive time step fails if t0 is an integer

### ISSUE THREAD

**Issue by macekmar** [netket/netket#1735]
Hi,

Adaptive time step integration fails if `t0` in `RungeKuttaIntegrator` is an integer (which for me is an easy mistake since `t0=0`).

Example code:
```python
import numpy as np
import netket.experimental.dynamics as nkd

def df(t, y, stage=None):
    return np.sin(t)**2*y

int_config = nkd.RK23(dt=0.04, adaptive=True, atol=1e-3, rtol=1e-3, dt_limits=[1e-3,1e-1])
integrator = int_config(df, 0, np.array([1.0]))  # <-- the second argument has to be a float

y_vec = [integrator.y]
t_vec = [0]
while integrator.t < 0.5:
    integrator.step()
    t_vec.append(integrator.t)
    y_vec.append(integrator.y)
```

An additional problem is that the traceback is confusing:
```
Traceback (most recent call last):
  File "/Users/macek/Work/Projects/TimeEvolution/DensityMatrix/script.py", line 13, in <module>
    integrator.step()
  File "/Users/macek/.pyenv/NetKet/lib/python3.12/site-packages/netket/experimental/dynamics/_rk_solver_structures.py", line 357, in step
    self._rkstate = self._do_step(self._rkstate, max_dt)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/macek/.pyenv/NetKet/lib/python3.12/site-packages/netket/experimental/dynamics/_rk_solver_structures.py", line 369, in _do_step_adaptive
    return general_time_step_adaptive(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/macek/.pyenv/NetKet/lib/python3.12/site-packages/netket/experimental/dynamics/_rk_solver_structures.py", line 46, in _maybe_jitted_fun
    res = fun(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^
  File "/Users/macek/.pyenv/NetKet/lib/python3.12/site-packages/netket/experimental/dynamics/_rk_solver_structures.py", line 249, in general_time_step_adaptive
    return jax.lax.cond(
           ^^^^^^^^^^^^^
  File "/Users/macek/.pyenv/NetKet/lib/python3.12/site-packages/netket/experimental/dynamics/_rk_solver_structures.py", line 151, in __repr__
    f", {self.last_norm:.2e}" if self.last_norm is not None else "",
        ^^^^^^^^^^^^^^^^^^^^
ValueError: Unknown format code 'e' for object of type 'str'
```
The last `ValueError` is from `RungeKuttaState.__repr__`, so quite confusing.

**Comment by PhilipVinc** [comment]
Thanks for raising the issue! 
Indeed, the repr method did not work well within `jax.jit` contexts, and you were seeing the error coming from there. 
I addressed both bugs in #1736 . Will be merged and tagged in a new release by tomorrow.

**Comment by PhilipVinc** [comment]
@macekmar NetKet 3.11.2 should address the issue.

Thanks a lot for the very clear reproducer, which makes it easy to quickly fix this kind of issues.
